File: urbanairship-core/src/test/java/com/urbanairship/analytics/SearchEventTemplateTest.java
/* Copyright Airship and Contributors */

package com.urbanairship.analytics;

import com.urbanairship.BaseTestCase;
import com.urbanairship.TestApplication;

import org.json.JSONException;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import java.math.BigDecimal;

import static org.mockito.Mockito.mock;

public class SearchEventTemplateTest extends BaseTestCase {

    @Rule
    public ExpectedException exception = ExpectedException.none();

    Analytics analytics;

    @Before
    public void setup() {
        analytics = mock(Analytics.class);
        TestApplication.getApplication().setAnalytics(analytics);
    }

    /**
     * Test basic Search Event with no optional value or properties.
     *
     * @throws JSONException
     */
    @Test
    public void testBasicSearchEvent() throws JSONException {
        CustomEvent event = SearchEventTemplate.newSearchTemplate().createEvent();

        EventTestUtils.validateEventValue(event, "event_name", "search");
        EventTestUtils.validateEventValue(event, "template_type", "search");
        EventTestUtils.validateNestedEventValue(event, "properties", "total_results", 0);
        EventTestUtils.validateNestedEventValue(event, "properties", "ltv", "false");
    }

    /**
     * Test Search Event with optional value and properties.
     *
     * @throws JSONException
     */
    @Test
    public void testSearchEvent() throws JSONException {
        CustomEvent event = SearchEventTemplate.newSearchTemplate()
                .setValue(new BigDecimal(123))
                .setType("FakeType")
                .setQuery("FakeQuery")
                .setCategory("FakeCategory")
                .setId("FakeID")
                .setTotalResults(23)
                .createEvent();

        EventTestUtils.validateEventValue(event, "event_name", "search");
        EventTestUtils.validateEventValue(event, "template_type", "search");
        EventTestUtils.validateEventValue(event, "event_value", 123000000L);
        EventTestUtils.validateNestedEventValue(event, "properties", "ltv", "true");
        EventTestUtils.validateNestedEventValue(event, "properties", "type", "FakeType");
        EventTestUtils.validateNestedEventValue(event, "properties", "query", "FakeQuery");
        EventTestUtils.validateNestedEventValue(event, "properties", "category", "FakeCategory");
        EventTestUtils.validateNestedEventValue(event, "properties", "id", "FakeID");
        EventTestUtils.validateNestedEventValue(event, "properties", "total_results", 23);
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/analytics/template/SearchEventTemplateTest.kt
/* Copyright Airship and Contributors */

package com.urbanairship.analytics.template

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.urbanairship.analytics.CustomEvent
import com.urbanairship.analytics.templates.SearchEventTemplate
import com.urbanairship.json.jsonMapOf
import junit.framework.TestCase.assertEquals
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class SearchEventTemplateTest {

    @Test
    public fun testSearch() {
        val event = CustomEvent.newBuilder(SearchEventTemplate.Type.SEARCH).build()

        assertEquals("search", event.eventName)
        assertEquals("search", event.templateType)
        assertEquals(jsonMapOf("ltv" to false), event.properties)
    }

    @Test
    public fun testProperties() {
        val event = CustomEvent.newBuilder(
            type = SearchEventTemplate.Type.SEARCH,
            properties = SearchEventTemplate.Properties(
                id = "some id",
                category = "some category",
                type = "some type",
                isLTV = true,
                query = "some query",
                totalResults = 20
            )
        ).build()

        assertEquals("search", event.eventName)
        assertEquals("search", event.templateType)
        assertEquals(jsonMapOf(
            "id" to "some id",
            "category" to "some category",
            "type" to "some type",
            "ltv" to true,
            "query" to "some query",
            "total_results" to 20
        ), event.properties)
    }
}
